src/RestrictedTokenFaucet.t.sol

```diff
@@ -0,0 +1,160 @@
+pragma solidity ^0.5.4;
+
+import "ds-test/test.sol";
+import "ds-token/token.sol";
+
+import "./RestrictedTokenFaucet.sol";
+
+contract FaucetUser {
+    DSToken token;
+    RestrictedTokenFaucet faucet;
+
+    constructor(DSToken token_, RestrictedTokenFaucet faucet_) public {
+        token = token_;
+        faucet = faucet_;
+    }
+
+    function doGulp() public {
+        faucet.gulp(address(token));
+    }
+
+    function doBalanceOf() public view returns(uint) {
+        return token.balanceOf(address(this));
+    }
+
+    function doUnDone(address usr) public {
+        faucet.unDone(usr, address(token));
+    }
+
+    function doRely(address usr) public {
+        faucet.rely(usr);
+    }
+
+    function doDeny(address usr) public {
+        faucet.deny(usr);
+    }
+
+    function doShut() public {
+        faucet.shut(ERC20Like(address(this)));
+    }
+
+    function doSetAmt(uint amt) public {
+        faucet.setamt(amt);
+    }
+}
+
+contract RestrictedTokenFaucetTest is DSTest {
+    RestrictedTokenFaucet faucet;
+    DSToken token;
+    address user1;
+    address user2;
+    address self;
+
+    function setUp() public {
+        faucet = new RestrictedTokenFaucet(20 ether);
+        token = new DSToken("TEST");
+        token.mint(address(faucet), 1000000 ether);
+        user1 = address(new FaucetUser(token, faucet));
+        user2 = address(new FaucetUser(token, faucet));
+        self = address(this);
+    }
+
+    function testSetupPrecondition() public {
+        assertEq(faucet.wards(self), 1);
+        assertEq(faucet.owner(), self);
+        assertEq(faucet.wards(user1), 0);
+        assertEq(faucet.wards(user2), 0);
+    }
+
+    function testFail_gulp_no_auth() public {
+        FaucetUser(user2).doGulp();
+    }
+
+    function test_gulp_auth() public {
+        faucet.rely(user1);
+        assertEq(faucet.wards(user1), 1);
+        assertEq(FaucetUser(user1).doBalanceOf(), 0);
```

Comment:
I feel like this is a bit unnecessary and you can just call `token.balanceOf(usr)`